File: converted/utilities/kebabcase/elide-kebabcase.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-kebabcase
# This demonstrates calling the TypeScript kebab-case implementation from Ruby.
# Benefits: Consistent URL slug generation across TypeScript and Ruby services.

puts "=== Ruby Calling TypeScript kebab-case ===\n"

# kebab_case = Elide.require('./elide-kebabcase.ts')

# Example: Rails URL Slug Generation
# class Post < ApplicationRecord
#   before_save :generate_slug
#
#   private
#   def generate_slug
#     self.slug = kebab_case.default(title)
#   end
# end
#
# post = Post.create(title: "Hello World Post")
# # post.slug => "hello-world-post"

# Example: CSS Class Helper
# module ApplicationHelper
#   def component_class(name)
#     "component-#{kebab_case.default(name)}"
#   end
# end
#
# component_class("UserProfile")  # "component-user-profile"

puts "Real-world use cases:"
puts "- Rails URL slug generation"
puts "- CSS class helper methods"
puts "- File name generation for static assets"
puts "- API endpoint naming conventions"
puts ""

puts "When Elide Ruby API is ready:"
puts "  kebab_case = Elide.require('./elide-kebabcase.ts')"
puts "  kebab_case.default('HelloWorld')  # 'hello-world'"

File: converted/utilities/kebabcase/elide-kebabcase.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-kebabcase
This demonstrates calling the TypeScript kebab-case implementation from Python.
Benefits: Consistent URL slug generation across TypeScript and Python services.
"""

# from elide import require
# kebab_case = require('./elide-kebabcase.ts')

print("=== Python Calling TypeScript kebab-case ===\n")

# Example 1: URL Slug Generation
# def create_slug(title):
#     """Generate URL-safe slug"""
#     return kebab_case.default(title)
#
# post_title = "Hello World Post"
# slug = create_slug(post_title)
# # Result: 'hello-world-post'

# Example 2: CSS Class Name Generation
# def generate_css_class(component_name):
#     """Generate BEM-style CSS class"""
#     return f"component-{kebab_case.default(component_name)}"
#
# class_name = generate_css_class("UserProfile")
# # Result: 'component-user-profile'

# Example 3: File Name Generation
# def generate_filename(display_name):
#     """Generate kebab-case filename"""
#     return f"{kebab_case.default(display_name)}.html"
#
# filename = generate_filename("Product Details")
# # Result: 'product-details.html'

print("Real-world use cases:")
print("- URL slug generation for blog posts/products")
print("- CSS class name generation (BEM style)")
print("- File name generation for static sites")
print("- API endpoint naming")
print()

print("When Elide Python API is ready:")
print("  kebab_case = require('./elide-kebabcase.ts')")
print("  kebab_case.default('HelloWorld')  # 'hello-world'")
